proto/api.proto

```diff
@@ -233,6 +233,7 @@ message EndTransactionRequest {
   // internal use only and will be ignored if requested through the
   // public-facing KV API.
   optional SplitTrigger split_trigger = 3;
+  optional MergeTrigger merge_trigger = 4;
```

Comment:
Note that you'll need to add a check to kv/db.go for merge_trigger as well. Follow the same format as what we're doing for split_trigger.